__tests__/helpers/setup-gha-env.ts

```diff
@@ -0,0 +1,40 @@
+import * as isCI from '../../src/lib/isCI';
+
+/**
+ * A helper function for setting up tests for simulating a GitHub Actions runner environment
+ */
+export function before() {
+  // List of all GitHub Actions env variables:
+  // https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables
+  process.env.GITHUB_ACTION = '__repo-owner_name-of-action-repo';
+  process.env.GITHUB_ACTIONS = 'true';
+  process.env.GITHUB_REPOSITORY = 'octocat/Hello-World';
+  process.env.GITHUB_RUN_ATTEMPT = '3';
+  process.env.GITHUB_RUN_ID = '1658821493';
+  process.env.GITHUB_RUN_NUMBER = '3';
+  process.env.GITHUB_SERVER_URL = 'https://github.com';
+  process.env.GITHUB_SHA = 'ffac537e6cbbf934b08745a378932722df287a53';
+  process.env.TEST_RDME_CI = 'true';
+
+  const isGHASpy = jest.spyOn(isCI, 'isGHA');
+  isGHASpy.mockReturnValue(true);
+
+  const ciNameSpy = jest.spyOn(isCI, 'ciName');
+  ciNameSpy.mockReturnValue('GitHub Actions (test)');
+}
+
+/**
+ * A helper function for tearing down tests after simulating a GitHub Actions runner environment
+ */
+export function after() {
+  delete process.env.GITHUB_ACTION;
+  delete process.env.GITHUB_ACTIONS;
+  delete process.env.GITHUB_REPOSITORY;
+  delete process.env.GITHUB_RUN_ATTEMPT;
+  delete process.env.GITHUB_RUN_ID;
+  delete process.env.GITHUB_RUN_NUMBER;
+  delete process.env.GITHUB_SERVER_URL;
+  delete process.env.GITHUB_SHA;
```

Comment:
Is deleting these from within our suite that's running on GH going to cause any problems? Should you reset these to what they were before the overwrites in the `before()` step?

Reply:
Hmm so we've been doing this in our main testbed for a while now and I haven't noticed any issues:
https://github.com/readmeio/rdme/blob/e3c7f22352d6f7e3d24c7fa762df788c905f25e4/__tests__/lib/fetch.test.ts#L29-L35

OH what if that's what's preventing us from writing jest coverage properly 👀 let me play around!

Reply:
alas that didn't fix the issues we were seeing in #617 😞 but this approach is a lot cleaner and tests still pass so I'll stick with it: https://github.com/readmeio/rdme/pull/735/commits/eafd73d9be5e4168bedc8f863039d8139af5e51b